### PR TITLE
Code clean up and added storage file format version on write check

### PR DESCRIPTION
**Pull request by Onager**

### CODE REVIEW

File: plaso/storage/interface.py [hunk 1]
```diff
@@ -698,6 +759,27 @@ def __init__(self, path):
     self._path = path
     self._storage_file = None
 
+  @property
+  def format_version(self):
+    """int: format version"""
```
Comment (joachimmetz) [inline plaso/storage/interface.py]: please add something in line of `or None if not set`

File: plaso/storage/sqlite/sqlite_file.py [hunk 2]
```diff
@@ -194,6 +197,10 @@ def _CheckStorageMetadata(cls, metadata_values):
     except (TypeError, ValueError):
       raise IOError('Invalid format version: {0!s}.'.format(format_version))
 
+    if not check_readable_only and format_version != cls._FORMAT_VERSION:
```
Comment (joachimmetz) [inline plaso/storage/sqlite/sqlite_file.py]: why not use something like `check_writable` and reverse the logic?

Comment (Onager) [reply plaso/storage/sqlite/sqlite_file.py]: This was to match the "read_only" behavior in the storage interface.

File: plaso/storage/interface.py (hunk 1 above)
Comment (Onager) [reply plaso/storage/interface.py]: Done

**Pull request merged**